tracer/build/_build/Build.Steps.cs

```diff
@@ -2293,4 +2204,41 @@ static string PrettyPrint(long bytes)
     }
 
     private record ParsedLogLine(DateTimeOffset Timestamp, LogLevel Level, string Message, AbsolutePath FileName);
+
+    private void DotnetBuild(
+        Project project,
+        MSBuildTargetPlatform platform = null,
+        TargetFramework framework = null,
+        bool noRestore = true,
+        bool noDependencies = true,
+        bool setNuget = false,
+        bool setPackages = false
+        )
+    {
+        DotnetBuild(new [] { project.Path }, platform, framework, noRestore, noDependencies, setNuget, setPackages); 
+    }
+
+    private void DotnetBuild(
+        IEnumerable<AbsolutePath> projPaths,
+        MSBuildTargetPlatform platform = null,
+        TargetFramework framework = null,
+        bool noRestore = true,
+        bool noDependencies = true,
+        bool setNuget = false,
+        bool setPackages = false
+        )
+    {
+        DotNetBuild(s => s
+            .SetConfiguration(BuildConfiguration)
+            .SetTargetPlatform(platform ?? MSBuildTargetPlatform.MSIL)
+            .When(noRestore, settings => settings.EnableNoRestore())
+            .When(noDependencies, settings => settings.EnableNoDependencies())
+            .When(framework is not null, settings => settings.SetFramework(framework))
+            .When(setNuget && !string.IsNullOrEmpty(NugetPackageDirectory), o => o.SetPackageDirectory(NugetPackageDirectory))
+            .When(setPackages && TestAllPackageVersions, o => o.SetProperty("TestAllPackageVersions", "true"))
+            .When(setPackages && IncludeMinorPackageVersions, o => o.SetProperty("IncludeMinorPackageVersions", "true"))
+            .SetNoWarnDotNetCore3()
+            .SetProcessArgumentConfigurator(arg => arg.Add("/nowarn:NU1701")) //nowarn:NU1701 - Package 'x' was restored using '.NETFramework,Version=v4.6.1' instead of the project target framework '.NETCoreApp,Version=v2.1'.
+            .CombineWith(projPaths, (settings, projPath) => settings.SetProjectFile(projPath)));        
```

Comment:
Just noting that this is _subtly_ different to the way it works with MSBuild. With MSBuild, it works out depedency order between `projPaths` and automatically builds them in the correct order, and in parallel. In this case we build the projects sequentially, in whatever order they're presented. It's not _actually_ an issue, because we "manually" build all dependencies first, just pointing it out for others 🙂